paho/pinger.go

```diff
@@ -109,13 +111,20 @@ func (p *DefaultPinger) Run(ctx context.Context, conn net.Conn, keepAlive uint16
 				timer.Reset(pingDue.Sub(t))
 				continue
 			}
-
-			lastPingSent = time.Now() // set before sending because WriteTo may return after PINGRESP is handled
-			if _, err := packets.NewControlPacket(packets.PINGREQ).WriteTo(conn); err != nil {
-				p.debug.Printf("DefaultPinger packet write error: %v", err)
-				return fmt.Errorf("failed to send PINGREQ: %w", err)
-			}
+			lastPingSent = time.Now()
+			go func() {
```

Comment:
Think it would be work mentioning why this has been done and the potential issue (have seen this issue logged against other clients). I don't think the spec really covers this... e.g.
// WriteTo may not complete within keepalive period so send in goroutine. Note potential issue if huge message is sent over very slow link at the same time as keepalive (so Write is blocked by other message and keepalive is exceeded). This is seems fairly unlikely and there is no obvious way to avoid the issue.